.github/workflows/coverage.yml

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+
+# https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+
+name: delta coverage checks
+
+on:
+  push:

```

Comment:
This will only run once the pull request is merged into the `main` branch.

Unfortunately I could not find a better way than duplicating the paths for the `pull_request` event:
https://github.com/intel/fpga-runtime-for-opencl/blob/d78bdb41d2d84a42f511c14bb8fe42c1c061effe/.github/workflows/build-linux.yml#L17-L48

Reply:
> This will only run once the pull request is merged into the `main` branch.

You mean the docker pull will face permission issue until it is merged into main?

Reply:
Good point. The pull request will run fine before being merged, but you need to include `package: read` permissions to pull the container image, which are private since these are only intended to be used in the CI workflows.
https://github.com/intel/fpga-runtime-for-opencl/blob/d78bdb41d2d84a42f511c14bb8fe42c1c061effe/.github/workflows/build-linux.yml#L8-L15

Reply:
Thanks you! It worked! 

Sorry I didn't notice this from the other files

Reply:
No worries :slightly_smiling_face: I made the same mistake in the first version of https://github.com/intel/fpga-runtime-for-opencl/pull/16, **after** telling you about package permissions. :facepalm:

Reply:
Actually you want to remove the `push` event, and only trigger on `pull_request`. For the `push` event there is no base ref so nothing to compare to; we could generate a report for the main branch, but that is a separate workflow for later.